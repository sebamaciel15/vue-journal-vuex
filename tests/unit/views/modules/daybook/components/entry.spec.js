import {shallowMount} from '@vue/test-utils';
import Entry from '@/modules/daybook/components/Entry.vue'
import {journalState} from '../../../mock-data/test-journal-state'


describe('pruebas en entry component', () => {
    //mockRouter
    const mockRouter = {
        push: jest.fn(),
    }
    //const wrapper = shallowMount(Entry, {... ...props ...global})
    const wrapper = shallowMount(Entry, {
        props: {
            entry:journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    })
    test('debe hacer match con el snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
    test('debe redireccionar al hacer click en el entry-container', () => {
        const entryContainer = wrapper.find('.entry-container')
        entryContainer.trigger('click')
        expect(mockRouter.push).toHaveBeenCalledWith({
            name: 'entry',
            params: {
                id: journalState.entries[0].id
            }
        })
    })
    test('pruebas en las propiedades computadas', () => {
        //wrapper.vm. <---- ver las propiedades computadas
        //day:23
        //month: julio
        //yearDay: '2021, viernes'
        expect(wrapper.vm.day).toBe(3)
        expect(wrapper.vm.month).toBe('Septiembre')
        expect(wrapper.vm.yearDay).toBe('2021, MiÃ©rcoles ')
    })
    
    
        
})
