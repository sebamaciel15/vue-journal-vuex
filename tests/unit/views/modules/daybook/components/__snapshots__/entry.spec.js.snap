// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en entry component debe hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- Titulo -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">3</span><span class="mx-1 fs-5">Septiembre</span><span class="mx-2 fw-light">2021, Mi√©rcoles </span></div>
  <div class="entry-description">Hola mundo desde Mock data</div>
</div>
`;
